### PR TITLE
feature: Scheduler signs type L1toL2 transactions

### PR DESCRIPTION
**Pull request by aodhgan**
as per https://github.com/mantlenetworkio/mantle/issues/378

### CODE REVIEW

File: l2geth/consensus/consensus.go [hunk 1]
```diff
@@ -59,6 +59,9 @@ type Engine interface {
 	// TODO Add comment
 	SetBatchPeriod(bps *types.BatchPeriodStartMsg)
 
+	// returns the current scheduler address
+	GetSchedulerAddress() (common.Address, error)
```
Comment (aodhgan) [inline l2geth/consensus/consensus.go:63]: @curryxbo What is the best/correct way to get the address/public key of the scheduler blockchain validation/ApplyTransaction?

File: scheduler/flags/flags.go [hunk 2]
```diff
@@ -206,6 +206,7 @@ var (
 		Value:  1,
 		EnvVar: prefixEnvVar("BLOCK_OFFSET"),
 	}
+	// is this the scheduler private key? why in optional flags?
```
Comment (aodhgan) [inline scheduler/flags/flags.go:209]: As far as I can tell, a Scheduler private key does not exist as a flag - but the Scheduler in the `consensus/clique` package has a Wallet field

File: l2geth/consensus/clique/scheduler.go [hunk 3]
```diff
@@ -134,6 +134,12 @@ func (schedulerInst *Scheduler) schedulerRoutine() {
 			Signature:    mockSignature,
 		}
 
+		// index block and sign L1toL2Tx here?
```
Comment (aodhgan) [inline l2geth/consensus/clique/scheduler.go:137]: Should signing happen here or in the `scheduler` package? i.e. `scheduler/drivers/scheduler/driver.go` etc

File: l2geth/consensus/consensus.go (hunk 1 above)
Comment (aodhgan) [reply l2geth/consensus/consensus.go:63]: I see in this PR (https://github.com/mantlenetworkio/mantle/pull/407/files) the scheduler address is available in l2geth/consensus/clique/scheduler.go Scheduler struct.

Comment (curryxbo) [reply l2geth/consensus/consensus.go:63]: @aodhgan Due to some changes to the schema, the scheduler unified production blocks, so this signature is now unnecessary，so the Issue will be closed

**Pull request closed without merging**